### ISSUE TITLE
Block redirects and let user choose whether to proceed or not

### ISSUE THREAD

**Issue by luxoflux** [gorhill/uMatrix#20]
I'm copying issue [#200](https://github.com/gorhill/httpswitchboard/issues/200) over from HTTPSB.

Basically, we need a way to intercept redirects, and have options to allow or deny, whitelist, blacklist, or what have you (up for discussion), with the ultimate aim of putting it under the user's control and consent.

“Ideally the extension should redirect to a blank page with a link and inform the user a page is trying to redirect.” <sup>[[1]](http://www.wilderssecurity.com/threads/umatrix-the-http-switchboard-successor.369601/page-2#post-2421841)</sup>

**Comment by WagnerGMD** [comment]
Right now, the ultimate aim seem's far away... The blank page has only one control : one close button...
In fact, it will be nice to add some options (now). A (simple) way to disable uMatrix ? By a right click (on the uMatrix icon), the user will have the choice (one tab or fully ?).
I assume, many users don't want to create a rule each time they met a redirect.

Otherwise, you could also think to add one feature like a redirect cleaner ? Maybe you could clean the redirect and provide a link ? And add this (clean) link inside the inform ?

**Comment by CrisBRM** [comment]
I suppose a toggle under 'strict https' would be nice addition since it's still bound by the scopes.

**Comment by WagnerGMD** [comment]
Since the v0.19pre0, [Policeman](https://github.com/futpib/policeman/releases/) has introduced a way to handle a redirect. You should maybe take a look. Maybe you don't have any idea to add such feature ? From my point of view, it's a good way (or a source of inspiration).

**Comment by gorhill** [comment]
> Maybe you don't have any idea to add such feature ?

Yes I have an idea, and no I do not need to look at anybody else's code.

**Comment by WagnerGMD** [comment]
It seem's that you made a mistake. Because it's just a question and a suggestion (nothing else, nothing more). It's normal to ask a question or to try to understand.

**Comment by ldgbc** [comment]
For references, RequestPolicyContinue support redirect blocking.

**Comment by WoLpH** [comment]
Is there any progress on this? It's quite annoying that I have to create a new rule for these sites and/or disable uMatrix completely for these urls to work. I don't want to enable these sites forever, but when they're the 1st party/redirect host, they should get whitelisted or even better, an "allow once" button.

To test, just try any of the product links on this site: https://www.1dagskoopjes.nl/

They all result in the following for me:
```
This webpage is not available

ERR_ADDRESS_UNREACHABLE
```

**Comment by polyzen** [comment]
@WoLpH, [RefControl](http://www.stardrifter.org/refcontrol/) does the job well. Will have to take a look at RequestPolicyContinued..

**Comment by ldgbc** [comment]
It seem uMatrix does support redirect control now but it require using Filter list rather than "on-the-fly" method. 

For reference, @polyzen mention RefControl: Note that RefControl just allow you to modify URL header (which URL you click on to open this page) and is not relate to this issue.

**Comment by RyanTG** [comment]
I just switched from RequestPolicy Continued, and I'm missing the ability to allow/block redirects. I'd love to see that added to uMatrix. Otherwise, uMatrix is great! The UI is well done. Thanks, folks.

**Comment by Roman-Nopantski** [comment]
For those missing the ability to allow/block redirects, until/if this feature is added to uMatrix, I recommend NoRedirect ( https://addons.mozilla.org/en-US/firefox/addon/noredirect/ ). It's pretty straightforward. It applies rules from the top down. And you can use Regex if you want. From Options in Add-ons panel, add `.*` as "Source" as your bottom rule (this is a default deny), then add domains above it as "Allow" eg I have `*arstechnica.com*` and other news sites that redirect from an rss feeds in my RSS newsreader.

**Comment by AtTheGates** [comment]
There is also the extension **Redirect Control** (https://addons.mozilla.org/firefox/addon/redirect-control), which is very similar to the redirect feature in **RequestPolicy** (**Continued**). It works very good in combination with **uMatrix** for me.
